Source/ARTJsonLikeEncoder.m

```diff
@@ -479,6 +496,50 @@ - (NSDictionary *)tokenDetailsToDictionary:(ARTTokenDetails *)tokenDetails {
     return dictionary;
 }
 
+- (NSDictionary *)deviceDetailsToDictionary:(ARTDeviceDetails *)deviceDetails {
+    NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+
+    dictionary[@"id"] = deviceDetails.id;
+    dictionary[@"platform"] = deviceDetails.platform;
+    dictionary[@"formFactor"] = ARTDeviceFormFactorToStr(deviceDetails.formFactor);
+
+    if (deviceDetails.clientId) {
+        dictionary[@"cliendId"] = deviceDetails.clientId;
+    }
+
+    dictionary[@"push"] = [self devicePushDetailsToDictionary:deviceDetails.push];
+
+    return dictionary;
+}
+
+- (ARTDeviceDetails *)deviceDetailsFromDictionary:(NSDictionary *)input error:(NSError * __autoreleasing *)error {
+    [_logger verbose:@"RS:%p ARTJsonLikeEncoder<%@>: deviceDetailsFromDictionary %@", _rest, [_delegate formatAsString], input];
+
+    if (![input isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+
+    ARTDeviceDetails *deviceDetails = [[ARTDeviceDetails alloc] initWithId:[input artString:@"id"]];
+    deviceDetails.updateToken = (ARTUpdateToken *)[input artString:@"updateToken"];
+    deviceDetails.push.state = ARTDevicePushStateFromStr([input artString:@"state"]);
+
+    return deviceDetails;
+}
+
+- (NSDictionary *)devicePushDetailsToDictionary:(ARTDevicePushDetails *)devicePushDetails {
+    NSMutableDictionary *dictionary = [NSMutableDictionary dictionary];
+
+    dictionary[@"transportType"] = devicePushDetails.transportType;
+
+    if (devicePushDetails.deviceToken) {
+        // Normalizing token by removing symbols and spaces
```

Comment:
Ah, sorry, I was wrong, this is what I did: `[deviceToken hexString]` using some random `hexString` method I found googling. So yes, removing symbols and spaces should do the same. Sorry!